apps/web/src/components/PaginatedTable/helpers/transactions.tsx

```diff
@@ -0,0 +1,187 @@
+import { EtherUnitDisplay } from "~/components/Displays/EtherUnitDisplay";
+import { Link } from "~/components/Link";
+import { RollupIcon } from "~/components/RollupIcon";
+import { Table } from "~/components/Table";
+import type { DeserializedBlob, DeserializedFullTransaction } from "~/utils";
+import {
+  buildAddressRoute,
+  buildBlockRoute,
+  formatBytes,
+  formatTimestamp,
+  shortenAddress,
+} from "~/utils";
+
+type Transaction = Pick<
+  DeserializedFullTransaction,
+  | "hash"
+  | "from"
+  | "to"
+  | "blobs"
+  | "rollup"
+  | "blockNumber"
+  | "blobGasBaseFee"
+  | "blobGasMaxFee"
+  | "blockTimestamp"
+> & { blobsLength?: number };
+
+export const transactionsTableHeaders = [
+  {
+    cells: [
+      {
+        item: "Hash",
+      },
+      {
+        item: "Block number",
+      },
+      {
+        item: "Timestamp",
+      },
+      {
+        item: "From",
+      },
+      {
+        item: "To",
+      },
+      {
+        item: "Rollup",
+      },
+      {
+        item: "Blob Base Fee",
+      },
+      {
+        item: "Blob Max Fee",
+      },
+      {
+        item: "Blob Gas Price",
+      },
+    ],
+  },
+];
+
+const getTransactionsTableRowExpandItem = (
+  blobs?: Pick<
+    DeserializedBlob,
+    "versionedHash" | "commitment" | "size" | "dataStorageReferences" | "proof"
+  >[]
+) => {
+  const headers = [
+    {
+      cells: [
+        {
+          item: "Blob Versioned Hash",
+        },
+        {
+          item: "Size",
+        },
+      ],
+      className: "dark:border-border-dark/20",
+      sticky: true,
+    },
+  ];
+
+  const rows = blobs
+    ? blobs.map((b) => ({
+        cells: [
+          {
+            item: b.versionedHash,
+          },
+          {
+            item: (
+              <div className="flex gap-2 text-xs">
+                <span>{formatBytes(b.size)}</span>
+              </div>
+            ),
+          },
+        ],
+        className: "dark:border-border-dark/10",
+      }))
+    : undefined;
+
+  return (
+    <Table
+      className="mb-4 mt-2 max-h-[420px] rounded-lg bg-primary-50 px-8 dark:bg-primary-800"
+      size="xs"
+      alignment="left"
+      variant="transparent"
+      headers={headers}
+      rows={rows}
+    />
+  );
+};
+
+export const getTransactionsTableRows = (transactions?: Transaction[]) =>
+  transactions
+    ? transactions.map((t: Transaction) => {
+        const {
+          hash,
+          from,
+          to,
+          blobs,
+          rollup,
+          blockNumber,
+          blobGasBaseFee,
+          blobGasMaxFee,
+          blockTimestamp,
+        } = t;
+        return {
+          cells: [
+            {
+              item: shortenAddress(hash, 6),
```

Comment:
Let's turn this into a link to the transaction details page

Reply:
Done :+1: